Allow consecutive runs of subctl join

mangelajo opened the pull request:
This will make sure that we update the Submariner operator resource in the
server

Fixes issue: #49

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/submariner-io/submariner-operator/79)
<!-- Reviewable:end -->

mangelajo commented:
@skitt sorry, I just realized this was assigned to you, ...

skitt commented on scripts/kind-e2e/e2e.sh:
```diff
@@ -258,7 +258,7 @@ for i in 2 3; do
                         --ikeport ${ce_ipsec_ikeport} \
                         --colorcodes ${subm_colorcodes} \
-			--disable-nat \
+			                  --disable-nat \
```
If you’re going to fix this, please use spaces throughout; there are still tabs at the start of the line ;-).

mangelajo replied:
ack, thanks

skitt commented on scripts/kind-e2e/e2e.sh:
```diff
@@ -311,6 +311,18 @@ for i in 2 3; do
+                --nattport ${ce_ipsec_nattport} \
+                --ikeport ${ce_ipsec_ikeport} \
+                --colorcodes ${subm_colorcodes} \
+                --disable-nat broker-info.subm
```
Excellent, I was going to ask for that!

skitt commented on pkg/subctl/operator/deploy/ensure.go:
```diff
@@ -42,24 +43,29 @@ func Ensure(config *rest.Config, submarinerSpec submariner.SubmarinerSpec) error
-	// } else if created {
-	//	fmt.Printf("* Created Submariner CR.\n")
-	// }
+	updateOrCreateSubmariner(submarinerClient, namespace, submariner)
```
This needs to be checked for errors.

mangelajo replied:
oh true, thanks

mangelajo commented:
@skitt , I addressed your comments, and also converted the comment about second time run for subctl into an echo because it was weird to see subctl being run again for "no specific reason".

mangelajo commented:
@skitt have an eye, comments addressed :-)

dfarrell07 commented:
> It failed the e2e tests for me, but I see it passing in CI so I will try again/debug in the morning. Looks good statically, non-blocker.

Re-running over the night, all passed/looks good in the morning. Thanks!

mangelajo commented on pkg/subctl/operator/deploy/ensure.go:
```diff
@@ -58,24 +59,33 @@ func Ensure(config *rest.Config, submarinerSpec submariner.SubmarinerSpec) error
+			return false, fmt.Errorf("failed to get pre-existing cfg %s : %s", submariner.Name, err)
+		}
+		submariner.ResourceVersion = existingCfg.ResourceVersion
+		_, err = clientSet.SubmarinerV1alpha1().Submariners(namespace).Update(submariner)
```
Oh, I didn't know we had those helpers ("CreateOrUpdate") I think we should switch to that lots of our code, thanks for the suggestion @tpantelis  I opened this bug: https://github.com/submariner-io/submariner-operator/issues/90
